samples/bluetooth/peripheral/src/main.c

```diff
@@ -279,12 +279,44 @@ BT_IAS_CB_DEFINE(ias_callbacks) = {
 	.high_alert = alert_high_start,
 };
 
+static int read_controller_info(void)
+{
+	struct net_buf *rsp;
+	int err;
+
+	/* Read Local Version Information */
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_LOCAL_VERSION_INFO, NULL, &rsp);
+	if (err) {
+		return err;
+	}
+
+	struct bt_hci_rp_read_local_version_info *ctlr_info = (void *)rsp->data;
+
+	printk("Controller information:\n");
+	printk("  status: 0x%02x\n", ctlr_info->status);
+	printk("  manufacturer: 0x%04x\n", ctlr_info->manufacturer);
```

Comment:
All 16-bit fields in the struct need sys_le16_to_cpu()

Reply:
thanks, don't know how I missed that.

Reply:
> thanks, don't know how I missed that.

In some ways it'd be nice if we had the same style of of byteorder-specific fixed width integer types/aliases like the Linux kernel: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/net/bluetooth/hci.h#n1355

Those are ultimately the same as `uint16_t` but since the byteorder is explicit you actually get compiler warnings if you try to directly pass that to an API which expects a CPU byteorder type or vice versa (e.g. trying to assign `__le16` to `uint16_t`).

Reply:
There was a PR that added these (https://github.com/zephyrproject-rtos/zephyr/pull/66920), but only for some static analysis tool it seems.